src/main/java/io/jenkins/pluginhealth/scoring/probes/LastCommitDateProbe.java

```diff
@@ -0,0 +1,78 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.File;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+import io.jenkins.pluginhealth.scoring.model.ResultStatus;
+
+import org.slf4j.Logger;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+
+/**
+ * Using the analysis done by {@link SCMLinkValidationProbe#}, this probe determines the last commit date of the plugin's repository.
+ */
+@Component
+@Order(value = LastCommitDateProbe.ORDER)
+public class LastCommitDateProbe extends Probe {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LastCommitDateProbe.class);
+
+    public static final int ORDER = SCMLinkValidationProbe.ORDER + 1;
+
+    @Override
+    public ProbeResult doApply(Plugin plugin) {
+        final String scmLinkValidationProbeKey = new SCMLinkValidationProbe(null, null).key();
+
+        try {
+            if (plugin.getDetails().get(scmLinkValidationProbeKey) == null) {
+                LOGGER.error("Couldn't run {} on {} because previous SCMLinkValidationProbe has null value in database", key(), plugin.getName());
+                return ProbeResult.error(key(), "SCM link has not been probed yet");
```

Comment:
shouldn't be `LOGGER.error` but `LOGGER.warning` and not a `ProbeResult.error` but `ProbeResult.failure`. Errors are not saved in DB.

Reply:
I thought that we wouldn't need to store anything for this probe in DB if the previous probe is null.
But I can see why that might be confusing for us in the future when we don't see any results for this probe in DB, would be difficult for us to debug what went wrong. But with `ProbeResult.failure` we'd capture this message in DB.

Sorry, I should have asked you this question in the Probe Engine PR, but when would you use `ProbeResult.failure` VS `ProbeResult.error`? Because in this case, I thought it was an error for the probe if the previous probe is null.

Reply:
You are correct. Let's keep it this way.